tests/cnf/ran/powermanagement/tests/cpufreq.go

```
@@ -0,0 +1,100 @@
+		Expect(err).ToNot(HaveOccurred(), "Failed to get performance profile")
+
+		By("get original isolated core frequency")
+		spokeCommand := fmt.Sprintf("cat /sys/devices/system/cpu/cpufreq/policy%v/scaling_max_freq |cat -",
+			isolatedCPUNumber)
+		consoleOut, err := cluster.ExecCommandOnSNO(raninittools.Spoke1APIClient, 3, spokeCommand)
+		Expect(err).ToNot(HaveOccurred(), "Failed to %s", spokeCommand)
+		freqAsInt, err := strconv.Atoi(strings.Trim(consoleOut, "\r\n"))
+		Expect(err).ToNot(HaveOccurred(), "strconv.Atoi Failed")
+		originalIsolatedCPUFreq = performancev2.CPUfrequency(freqAsInt)
+
+		By("get original reserved core frequency")
+		spokeCommand = fmt.Sprintf("cat /sys/devices/system/cpu/cpufreq/policy%v/scaling_max_freq |cat -",
+			ReservedCPUNumber)
+		consoleOut, err = cluster.ExecCommandOnSNO(raninittools.Spoke1APIClient, 3, spokeCommand)
+		Expect(err).ToNot(HaveOccurred(), "Failed to %s", spokeCommand)
+		freqAsInt, err = strconv.Atoi(strings.Trim(consoleOut, "\r\n"))
+		Expect(err).ToNot(HaveOccurred(), "strconv.Atoi Failed")
+		originalReservedCPUFreq = performancev2.CPUfrequency(freqAsInt)
+	})
+
+	AfterEach(func() {
+		By("Reverts the CPU frequencies to the original setting")
+		err = helper.SetCPUFreqAndWaitForMcpUpdate(perfProfile, *nodeList[0],
+			&originalIsolatedCPUFreq, &originalReservedCPUFreq)
+		Expect(err).ToNot(HaveOccurred(), "Failed to set CPU Freq")
+	})
+
+	FContext("Reserved Core Frequency Tuning Test", func() {
+		It("tests changing reserved and isolated CPU frequencies", func() {
+
+			By("patch performance profile to set core frequencies")
+			err = helper.SetCPUFreqAndWaitForMcpUpdate(perfProfile, *nodeList[0],
+				&desiredIsolatedCoreFreq, &desiredReservedCoreFreq)
+			Expect(err).ToNot(HaveOccurred(), "Failed to set CPU Freq")
+
+			By("Get modified isolated core frequency")
+			spokeCommand := fmt.Sprintf("cat /sys/devices/system/cpu/cpufreq/policy%v/scaling_max_freq |cat -",
```

why adding "| cat -" to mask cmd failure? It would be better to let the test fail when the actual issue is encountered. 

```
[kni@registry.kni-qe-72 ~]$ cat /sys/devices/system/cpu/cpufreq/policy%v/scaling_max_freq |cat -
cat: /sys/devices/system/cpu/cpufreq/policy%v/scaling_max_freq: No such file or directory
[kni@registry.kni-qe-72 ~]$ echo $?
0
[kni@registry.kni-qe-72 ~]$ cat /sys/devices/system/cpu/cpufreq/policy%v/scaling_max_freq
cat: /sys/devices/system/cpu/cpufreq/policy%v/scaling_max_freq: No such file or directory
[kni@registry.kni-qe-72 ~]$ echo $?
1
```

---

@klaskosk It will make retry pretty much useless if we append "| cat - ". How often do you encounter this issue? We might need to think of other solution to strip off the timestamp.